libs/ledger/src/fees/fee_manager.cpp

```diff
@@ -92,14 +84,12 @@ bool FeeManager::CalculateChargeAndValidate(TransactionDetails &             tx,
                   " (base: ", base_charge, " storage: ", storage_charge,
                   " compute: ", compute_charge, " shards: ", allowed_shards_.PopCount(), ")");
 
-  if (!tx.is_create_wealth)
-  {
-    result.charge += scaled_charge;
-  }
+  result.charge += scaled_charge;
 
   // determine if the chain code ran out of charge
   if (result.charge > tx.charge_limit)
   {
+    result.charge = scaled_charge;
```

Comment:
Not needed here. If the status is not success maximum fees are taken from the user

Reply:
ok